integration/aggregate_variables_compat_test.go

```diff
@@ -44,6 +44,16 @@ func TestAggregateVariablesCompatRoot(t *testing.T) {
 			failsForSQLite: "https://github.com/FerretDB/FerretDB/issues/3148",
 			skip:           "https://github.com/FerretDB/FerretDB/issues/1992",
 		},
+		"GroupIDTwice": {
+			pipeline: bson.A{
+				bson.D{{"$sort", bson.D{{"_id", 1}}}},
+				bson.D{{"$group", bson.D{{"_id", "$$ROOT"}}}},
+				bson.D{{"$group", bson.D{{"_id", "$$ROOT"}}}},
+				bson.D{{"$sort", bson.D{{"_id", 1}}}},
+			},
```

Comment:
$$ROOT uses document of its own pipeline stage, not the initial pipeline stage. It's rather strange example, found no other stage to demonstrate it. `$addFields` doesn't implement expression yet.
```
test> db.coll.insertOne({_id:1,v:1})                                                                                                                                                                        
{ acknowledged: true, insertedId: 1 }  
test> db.coll.aggregate([{ $group: { "_id": "$$ROOT" } }, { $group: { "_id": "$$ROOT" } }])           
[                                                                                                     
  { _id: { _id: { _id: 1, v: 1 } } }                                                                  
] 
```